src/deadline/job_attachments/api/manifest.py

```
@@ -247,3 +260,182 @@ def process_output(status: FileStatus, path: str, output_diff: ManifestDiff):
+    """
+    # Always upload the manifest file to case root /Manifest with the original file name.
+    manifest_path = manifest_path + "/Manifests/" + Path(manifest_file).name
+
+    # S3 uploader.
+    upload = S3AssetUploader(session=boto_session)
+    with open(manifest_file) as manifest:
+        upload.upload_bytes_to_s3(
+            bytes=BytesIO(manifest.read().encode("utf-8")),
+            bucket=s3_bucket_name,
+            key=manifest_path,
+            progress_handler=logger.echo,
+            extra_args=s3_metadata,
+        )
+
+
+def _manifest_download(
+    download_dir: str,
+    farm_id: str,
+    queue_id: str,
+    job_id: str,
+    boto3_session: boto3.Session,
+    step_id: Optional[str] = None,
+    logger: ClickLogger = ClickLogger(False),
+) -> ManifestDownloadResponse:
+    """
+    BETA API - This API is still evolving but will be made public in the near future.
+    API to download the Job Attachment manifest for a Job, and optionally dependencies for Step.
+    download_dir: Download directory.
+    farm_id: The Deadline Farm to download from.
+    queue_id: The Deadline Queue to download from.
+    job_id: Job Id to download.
+    boto_session: Boto3 session.
+    step_id: Optional[str]: Optional, download manifest for a step
+    logger: Click Logger instance to print to CLI as test or JSON.
+    return ManifestDownloadResponse Downloaded Manifest data. Contains source S3 key and local download path.
+    """
+
+    # Deadline Client and get the Queue to download.
+    deadline = boto3_session.client("deadline", config=get_default_client_config())
+
+    queue: dict = deadline.get_queue(
+        farmId=farm_id,
+        queueId=queue_id,
+    )
+
+    # assume queue role - session permissions
+    queue_role_session: boto3.Session = _get_queue_user_boto3_session(
+        deadline=deadline,
+        base_session=boto3_session,
+        farm_id=farm_id,
+        queue_id=queue_id,
+        queue_display_name=queue["displayName"],
```

Could be `queue.displayName`, same as other queue attribute data access.

---

This is a dict, and not the `get_queue` helper.

If the function does need all of the other properties, then we can use `get_queue` with the `Queue` structure. Just saving some code when not more than 50% is necessary

---

Chatted offline, I am against using helpers that swallow up the real error under the hood. Either we use generate boto3 stubs, or none at alll.